### ISSUE TITLE
Behavior of include totally changes in 2.1

### ISSUE THREAD

**Issue by windowsrefund** [ansible/ansible#15342]
##### ISSUE TYPE
- Bug Report
##### ANSIBLE VERSION

```
ansible 2.1.0 (devel 1f82c2c4a8) last updated 2016/04/08 10:55:03 (GMT -400)
```
##### CONFIGURATION

N/A
##### OS / ENVIRONMENT

N/A
##### SUMMARY

Work flow of include has changed in a drastic way between 2.0 and 2.1
##### STEPS TO REPRODUCE

```
- name: "Build a file of tasks to include"                                         
  template:                                                                        
    src: include_tasks.j2                                                          
    dest: "/tmp/include_tasks.yml"                                                 

- name: Run tasks against reference instances                                      
  include: "/tmp/include_tasks.yml"

```
##### EXPECTED RESULTS

The above work flow (dynamically build a set of tasks to run later) works on 2.0
##### ACTUAL RESULTS

The same code fails on 2.1. It seems the include: is run before any tasks in the play on 2.1.

```
 fatal: [localhost]: FAILED! => {"failed": true, "reason": "the file_name '/tmp/include_tasks.yml' does not exist, or is not readable"}
```

**Comment by dagwieers** [comment]
Hmm, what I see is that you are templating the file (to the remote target), but you are including the file on the master server (running ansible-playbook). I don't think this work with v2.0 or 1.9 either. If you want to make this work, you can do something like:

```yaml
- name: "Build a file of tasks to include"                                         
  template:                                                                        
    src: include_tasks.j2                                                          
    dest: "/tmp/include_tasks.yml"
    transport: local

- name: Run tasks against reference instances
  include: "/tmp/include_tasks.yml"
```

**Comment by jimi-c** [comment]
So really, in this situation, the include should be marked as `static: no` to enforce the fact that it should not be read in statically (and thus causing the error). Unfortunately, this would require you to modify your playbooks once 2.1 was released. I think for now we'll add a deprecation warning here, so that by 2.3 you'd need to set `static: no` for this to work properly.

**Comment by jimi-c** [comment]
# Closing This Ticket

Hi!

We believe the above commit should resolve this problem for you. This will also be included in the next release.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!

**Comment by geerlingguy** [comment]
I noticed that in 2.1.2 there's now a huge list of (by default) statically included files at the beginning of a playbook run (I don't recall seeing it in 2.1.1). Is there any way to ignore / not print that? And couldn't fairly substantial UX changes like that be made in minor releases and not patch?

**Comment by kustodian** [comment]
@geerlingguy check out this pull request #17918.

**Comment by jimi-c** [comment]
Merged #17918 (and CP'd to `stable-2.{1,2}`).

**Comment by geerlingguy** [comment]
Ugh, still hitting this in many of my roles, which I upgraded to assume 2.0/early 2.1 behavior. No I'm having to run through many of them and add `static: no` to select includes to fix this (see examples below):
- https://github.com/weareinteractive/ansible-newrelic/issues/13
- https://github.com/geerlingguy/ansible-role-java/issues/26
- https://github.com/geerlingguy/ansible-role-postgresql/issues/7
- https://github.com/geerlingguy/ansible-role-apache/issues/110
